catalog/ui/search-ui/search-endpoint/src/main/java/org/codice/ddf/ui/searchui/query/controller/SearchController.java

```diff
@@ -122,6 +122,7 @@ public synchronized void publishResults(
    * @param request {@link SearchRequest} containing a query for 1 or more sources
    * @param session Cometd {@link ServerSession}
    */
+  @SuppressWarnings("squid:S2142" /* Thread interrupt handled */)
```

Comment:
❗️ That doesn't cut it. Here we need to compute the end time and time remaining in each loop and make sure we wrap the whole thing within a loop where we can propagate the interruption in a finally clause outside of that while loop